### PR TITLE
add filter for complete/incomplete todos

### PR DESCRIPTION
**Pull request by Swati4star**
Option to add 
`filter="complete"`
`filter="incomplete"`
to todo list

Another enhancement as mentioned in https://github.com/pimutils/todoman/issues/18

Example:

```
 todo list --filter="complete"
6  [X]    2017-02-22 07:21:35  HELLO @Personal (100%)
1  [X]    2017-02-22 05:35:56  HELLO @Personal (100%)

todo list --filter="incomplete"
 8  [ ]    2017-06-22             long task @Personal
 9  [ ]    Today 02:40:45         my @Personal
 7  [ ]    2017-02-22 11:08:17    hello task 2 @Personal
 5  [ ]    2017-02-22 10:49:22    do something @Personal
 4  [ ]    2017-02-22 10:48:54    Task 3 @Personal
 3  [ ]    2017-02-22 07:21:37    HELLO @Personal

```

### CODE REVIEW

**Comment by untitaker** [comment]
Oh, that part of #18 was rather brainstorming! We already have `--all`, so we need to think about how to keep this consistent.

**Comment by hobarrera** [comment]
Yeah, that suggestion was kinda dumped there and sort of old. I think it makes sense the most to add a `--completed` flag. Ideas:

* `--completed=[yes|no|all]` (I think each option is clear). Drop `--all`.
* `--completed` only shows completed ones (no arguments).

Opinion, @untitaker ?

@Swati4star Please hang on a bit while we decide exactly *how* we want to implement this. ^_^

**Comment by untitaker** [comment]
I prefer something like the second option, but with `--completed` renamed to `--completed-only`.

**Comment by untitaker** [comment]
Also we need to prevent the user from using `--all` AND `--completed-only`, as they contradict each other.

**Comment by hobarrera** [comment]
Sounds good, just one nit-pick: I like `--done` better. ;)

**Comment by untitaker** [comment]
Good, but please let's call it `--done-only`, otherwise it can be misinterpreted to behave the same as `--all`.

**Comment by hobarrera** [comment]
Sorry, @Swati4star, we made this a bit complicated since we hadn't really designed some bits as well. 😅

**Comment by Swati4star** [comment]
Now, I have added a flag --done-only, 
The output is like this,

```
todo list --done-only
6  [X]    2017-02-22 07:21:35  HELLO @Personal (100%)
1  [X]    2017-02-22 05:35:56  HELLO @Personal (100%)
```

File: todoman/model.py [hunk 1]
```diff
@@ -577,9 +580,13 @@ def todos(self, all=False, lists=[], urgent=False, location='',
 
         if not all:
             # XXX: Duplicated logic of Todo.is_completed
-            extra_where.append('AND completed_at is NULL '
-                               'AND status != "CANCELLED" '
-                               'AND status != "COMPLETED"')
+            if complete:
+                extra_where.append('AND status != "CANCELLED" '
+                                   'AND status == "COMPLETED"')
```
Comment (hobarrera) [inline todoman/model.py]: `status != "CANCELLED"` isn't necessary; anything matching `status == "COMPLETED` will already be not-cancelled (can't have two statuses :P ).

**Comment by untitaker** [comment]
Also it appears that your new test currently fails

**Comment by Swati4star** [comment]
screenshot from 2017-02-26 19 42 14

@untitaker @hobarrera Can you please tell why am I getting this error, when I am running tox.

**Comment by untitaker** [comment]
You need to install todoman again (`pip install -e .`) because we have added a
new dependency in the meantime.

**Comment by Swati4star** [comment]
@untitaker Fixed the tests! :+1:

**Comment by untitaker** [comment]
Fantastic, thanks!

**Comment by hobarrera** [comment]
Thanks!

**Pull request merged**